test/one/profiler/test/RunnerDeclaration.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package one.profiler.test;
+
+import java.lang.reflect.Method;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class RunnerDeclaration {
+    private static final Logger log = Logger.getLogger(RunnerDeclaration.class.getName());
+
+    private final Set<String> includeDirs;
+    private final List<Pattern> filters;
+    private final Set<String> looksLikeClassName;
+    private final Set<String> looksLikeMethodName;
+
+    private final List<Pattern> skipFilters;
+    private final Set<String> skipLooksLikeClassName;
+    private final Set<String> skipLooksLikeMethodName;
+
+    public RunnerDeclaration(List<String> includeDirs, List<String> globs, List<String> skipGlobs) {
+        this.includeDirs = new HashSet<>(includeDirs);
+        this.filters = globs.stream().map(Filter::from).collect(Collectors.toList());
+        this.looksLikeClassName = globs.stream().filter(f -> !f.contains(".") && !f.contains("*") && Character.isUpperCase(f.charAt(0))).collect(Collectors.toSet());
+        this.looksLikeMethodName = globs.stream().filter(f -> !f.contains(".") && !f.contains("*") && Character.isLowerCase(f.charAt(0))).collect(Collectors.toSet());
```

Comment:
I think all these 4 structures can be replaced with a single list of patterns as follows:
1. glob is a test dir => `GlobTests.*`
2. glob looks like a class name => `Glob.*`
3. glob looks like a method name => `*.glob`
4. otherwise => `glob`